aten/src/ATen/native/native_functions.yaml

```diff
@@ -4526,26 +4526,14 @@
     CPU: channel_shuffle_cpu
     CompositeImplicitAutograd: math_channel_shuffle
 
-- func: is_pinned(Tensor self, Device? device=None) -> bool
+- func: is_pinned(Tensor self) -> bool
```

Comment:
This is going to be badly BC-breaking I'm afraid.
To make this PR landing more easily, I would preserve the device arg here and continue to respect it. Only change the default to point to the accelerator.

We can then have 3 follow ups:
- Update all pt callers not to pass this arg
- Throw a deprecation warning if this arg is passed
- After 2 releases, actually remove the arg.

Reply:
Agree with you.

- If low level implement of pin_memory is acceptable, I will go on for the next step to update all PT callers of related APIs.
- Have added deprecation warnings in C++ side. But I'm not sure if it's enough, WDYT?

Reply:
Hey,
Will take a look at the code.
By what I meant by 3 follow ups here meant that this can be 3 different PRs (doesn't have to be).

Reply:
Thanks!
As what you said, to make this PR landing more easily, this PR will only refactor implement of pin memory and add a deprecation warning for `device` arg. I will submit another PR to update all PT callers based on this PR.